core/src/stores/postgres.rs

```diff
@@ -2847,51 +2923,56 @@ impl Store for PostgresStore {
         Ok(())
     }
 
-    async fn upsert_data_source_folder(
-        &self,
-        project: &Project,
-        data_source_id: &str,
-        folder: &Folder,
-    ) -> Result<()> {
-        let project_id = project.project_id();
-        let data_source_id = data_source_id.to_string();
+    async fn upsert_data_source_folder(&self, folder: &Folder) -> Result<()> {
+        let data_source_id = folder.data_source_id();
 
         let pool = self.pool.clone();
-        let c = pool.get().await?;
+        let mut c = pool.get().await?;
 
-        let stmt = c
-            .prepare(
+        let created = utils::now();
+
+        // get the data source row id
+        let tx = c.transaction().await?;
+        let r = tx
+            .query(
                 "SELECT id FROM data_sources WHERE project = $1 AND data_source_id = $2 LIMIT 1",
+                &[&folder.project().project_id(), &data_source_id],
             )
             .await?;
-        let r = c.query(&stmt, &[&project_id, &data_source_id]).await?;
         let data_source_row_id: i64 = match r.len() {
             0 => Err(anyhow!("Unknown DataSource: {}", data_source_id))?,
             1 => r[0].get(0),
             _ => unreachable!(),
         };
 
-        let stmt = c
+        let stmt = tx
             .prepare(
                 "INSERT INTO data_sources_folders \
-               (id, data_source, created, folder_id) \
-               VALUES (DEFAULT, $1, $2, $3) \
-               ON CONFLICT (data_source, folder_id) DO UPDATE \
-               SET created = EXCLUDED.created \
-               RETURNING id",
+                       (id, data_source, created, folder_id) \
+                       VALUES (DEFAULT, $1, $2, $3) \
+                       ON CONFLICT (folder_id, data_source)  DO UPDATE \
+                       SET folder_id = data_sources_folders.folder_id \
```

Comment:
I guess it should be on conflict do nothing right ? 
Doesn't matter though I'm changing this slightly

Reply:
yes, but do nothing does not return the id .  SET folder_id = data_sources_folders.folder_id  does nothing, but returns something